internal/storage/image.go

```diff
@@ -873,3 +883,43 @@ func GetImageService(ctx context.Context, sc *types.SystemContext, store storage
 
 	return is, nil
 }
+
+// FilterPinnedImage checks if the give image needs to be pinned
+// to exclude from kubelet's image GC.
+func FilterPinnedImage(image string, pinnedImages []*regexp.Regexp) bool {
+	if len(pinnedImages) == 0 {
+		return false
+	}
+
+	for _, pinnedImage := range pinnedImages {
+		if pinnedImage.MatchString(image) {
+			return true
+		}
+	}
+	return false
+}
+
+// CompileRegexpsForPinnedImages compiles regular expressions for the given
+// list of pinned images.
+func CompileRegexpsForPinnedImages(patterns []string) []*regexp.Regexp {
+	regexps := make([]*regexp.Regexp, 0, len(patterns))
+	for _, pattern := range patterns {
+		var re *regexp.Regexp
+		switch {
```

Comment:
Hm, I’ll not block on this but I’d prefer having plain globs or regular expressions without any further checks.